### ISSUE TITLE
Formatting in constructor the initial value seems not to work as expected

### ISSUE THREAD

**Issue by RaitoBezarius** [razagill/react-phone-input#8]
I am trying to pass a `value` which is non-null to the component, I see that the constructor tries to format it.

But when I pass an already-formatted value like: `"+1 (562) 756-2233"`, the result is `formattedNumber: "+"` in the state.

I don't really have taken a look to the `formatNumber` algorithm, but supposedly it looks like that we should maybe sanitize more the number (like removing everything which is not a number: parenthesis, spaces).

I could land a PR also for this too, I suppose.

**Comment by sample-usr** [comment]
@RaitoBezarius Yes, seems we need some sanitization there. I am gonna be back to work in two weeks, but if you get time before that a PR is most welcome :)
